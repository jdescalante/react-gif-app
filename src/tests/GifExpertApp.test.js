import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'


describe('Pruebas en componente <GifExpertApp />', () => {
    
    test('debe de mostrar el componente GifExpertApp', () => {
        const wrapper = shallow( <GifExpertApp />);

        expect( wrapper ).toMatchSnapshot()

    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['Resident Evil', 'Marvel'];
        const wrapper = shallow( <GifExpertApp defaultCategory={ categories } />);
        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length).toBe( categories.length );
    })
    
    
})
