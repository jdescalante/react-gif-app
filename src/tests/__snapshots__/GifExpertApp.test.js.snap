// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifExpertApp /> debe de mostrar el componente GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Resident Evil"
      key="Resident Evil"
    />
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
  </ol>
</Fragment>
`;
